v2/middlewares/measurement.go

```diff
@@ -0,0 +1,69 @@
+package middlewares
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+
+	"github.com/honestbank/kp/v2/internal/middleware"
+)
+
+var operationDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "kp_operation_time_milliseconds",
```

Comment:
Although this works fine, in practice I have not see any metric in Prometheus to include time units. I have rather seen something like:
```suggestion
	Name:    "kp_operation",
```

And have `kp_operation`, in addition to labels such as `result`, have a label `durationMs` and even multiple such as `durationS`. But again, a bit up to you. But again, I think it's way better if we pack the duration information within a label together with other stuff, instead of a separate metric for the duration.

Reply:
Actually, I wanna add now: The name should now end up with `_time_milliseconds` as this is carrying the labels for errors etc.
So I encourage this metric to be named `kp_operation`

Reply:
![image](https://user-images.githubusercontent.com/3640284/200241279-ad4e86bd-1d1c-42ab-b69a-8f2f5940df27.png)
@jai had recommended taking a look at istio metrics, and following their standard because they are following proper conventions.

Apparently, we don't store the actual precise duration, but rather take the duration and Prometheus adds it to one of the buckets like in the below screenshot :)

![image](https://user-images.githubusercontent.com/3640284/200241460-6df93555-feb0-4edd-b46c-54f1cfc40cc8.png)

let me know if you feel strongly about this one, we'll need to re-visit if so :)